### ISSUE TITLE
Request: add Dockerfiles for samtools

### ISSUE THREAD

**Issue by magicDGS** [samtools/samtools#776]
It will be nice to have a repository in the samtools organization (or in the samtools repository) with Dockerfiles for samtools and related software (e.g., tabix) and upload docker images to DockerHub.

This will be useful to have official images for samtools that have the versioning of the repositor, and to have in every release a ready-to-go docker image to work with.

I recommend alpine-linux as base image to have it as small as possible, but that's not a requirement for me...

**Comment by dpryan79** [comment]
Is there a reason the docker and singularity images generated automatically by the bioconda project don't suffice?

**Comment by magicDGS** [comment]
The info of BioContainers (where in principle bioconda automatic uploads happen) is in version 1.3.1 (https://biocontainers.pro/registry/#/showImages?domain=dockerhub&repository=samtools&starred=false). When I went to the quay.io registry, I found that 1.7 is there, but it has a weird version name (1.7--0) instead of the expected 1.7

I would like to run, for example, `docker pull ${repository}/samtools:${version}`, where I don't really mind in which repository is found (`biocontainers`, `quay.io/biocontainers` `samtools`), but where the version is the one in the samtools releases. Like that, it is easier to get correct and official versions.

I submitted a PR to biocontainers to update the version and add a Dockerfile for the version 1.7, but I thought that it would be nice that the samtools organization release docker images at the same time as versions independently of biocontainers or coda.

I can understand that maybe is not a priority or it will increase the maintenance workload, so I am fine also if the request is not taken into account.
